numpy/lib/stride_tricks.py

```diff
@@ -111,6 +111,35 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
 
     return view
 
+def sliding_window_view(x, shape=None):
+    """
+    Create rolling window views of the 2D array with the given shape.
+
+    .. warning:: This function has to be used with extreme care, see notes from 'as_strided'.
```

Comment:
You should design this function so that this is not true. Perhaps just set the readonly flag on the returned array?

Reply:
You mean the warning? This warning comes from the  'as_strided', which I this function depends on. I don't know if it's a good idea to remove this warning?

Reply:
Yes, I mean the warning. This warning is aimed at _you_, the caller of `as_strided`. You should be the one taking extreme care, so that the caller of `sliding_window_view` does not have to

Reply:
Add shape and step_size check. Warning can be safety removed.